analyses/copy_number_consensus_call/Snakefile

```diff
@@ -0,0 +1,81 @@
+# Nhat Duong -- 
+# Updated Dec 5, 2019
+
+## Define the ending file(s) that we want
+ALL_FREEC= expand("../../scratch/interim/{sample}.freec.dup.bed", sample=config["samples"])
+ALL_CNVKIT= expand("../../scratch/interim/{sample}.cnvkit.dup.bed", sample=config["samples"])
+ALL_MANTA= expand("../../scratch/interim/{sample}.manta.dup.bed", sample=config["samples"])
+        
+
+## Define the first rule of the Snakefile. This rule determines what the final file is and which steps to be taken.
+rule all:
+    input:
+        ALL_FREEC + ALL_CNVKIT + ALL_MANTA
+
+
+#########################      FILTER STEP FOR THE 3 CALL METHODS ####################################
+
+
+rule freec_filter:
+    input:
+        ## Define the location of the input file and take the extension from the config file
+        events="../../scratch/freec_freec/{sample}" + str(config["freec_ext"])
+    output:
+        ## Define the output files' names
+        freec_del="../../scratch/interim/{sample}.freec.del.bed",
+        freec_dup="../../scratch/interim/{sample}.freec.dup.bed"
+    threads: 1
+    params:
+        ## Take parameters from the config file and assign them into params for convinient use in the Shell section
+        SIZE_CUTOFF=str(config["size_cutoff"]),
+        FREEC_PVAL=str(config["freec_pval"]),
+        mem=r'1G'
```

Comment:
What does this line `mem=...` do? The snakemake docs that I found have this in a resources section and use `mem_mb`

Reply:
This line specifies the **memory usage** for this specific task **on the cluster**. So for this task, Snakemake is to use 1G of memory to complete it on the cluster. One can make use of the `mem=...` and `threads` by adding
`-c "qsub -l h_vmem={params.mem} -l mem_free={params.mem} -pe smp {threads} -V -cwd"`
to the Snakemake command in the `run_consensus_call.sh` file

However, for this pull request, I designed this to work on the local system so I didn't include the `-c` flag in the `run_consensus_call.sh` file. Thus, the `men=...` is not being used